cvat/apps/engine/task.py

```diff
@@ -61,12 +62,34 @@ class SegmentsParams(NamedTuple):
     segment_size: int
     overlap: int
 
-def _copy_data_from_source(server_files, upload_dir, server_dir=None):
+def _copy_data_from_share_point(
+    server_files: List[str],
+    upload_dir: str,
+    server_dir: Optional[str] = None,
+    server_files_exclude: Optional[List[str]] = None,
+):
     job = rq.get_current_job()
     job.meta['status'] = 'Data are being copied from source..'
     job.save_meta()
 
-    for path in server_files:
+    filtered_server_files = server_files.copy()
+
+    # filter data from files/directories that should be excluded
+    if server_files_exclude:
+        for f in server_files:
+            path = Path(server_dir or settings.SHARE_ROOT) / Path(f)
+            if not path.is_dir():
+                filtered_server_files.append(f)
```

Comment:
This seems to add a duplicate element to `filtered_server_files` (since `filtered_server_files` is initialized from `server_files.copy()`, `f` must already be in it).